File: mhw/rollhunt.js
const Command = require( '../utils/baseCommand.js' )

const monsterDatabase = require('../utils/databases/mhw/monsters.json');
const weaponDatabase = require('../utils/databases/mhw/weapons.json');
const armorDatabase = require('../utils/databases/mhw/armors.json');

class RollHunt extends Command {
  constructor() {
    super(
      'rollhunt',
      'rollhunt',
      'Get a random roll of what monster you should hunt with which gear'
    )
  }

  run(client,message,args) {

    let input = args.join('').toLowerCase();

    const mhMonsterKeys = Object.values(monsterDatabase);
    const mhWeaponKeys = Object.values(weaponDatabase);
    const mhArmorKeys = Object.values(armorDatabase);

    const monster = mhMonsterKeys[Math.floor(Math.random() * mhMonsterKeys.length)].details.title;
    const weapon = mhWeaponKeys[Math.floor(Math.random() * mhWeaponKeys.length)].title;
    const armor = mhArmorKeys[Math.floor(Math.random() * mhArmorKeys.length)].name;

    message.channel.send("Fight " + "**" + monster + "**" + " with a " + "**" + weapon + "**" + " using " + "**" + armor + "**");
  }
}

module.exports = RollHunt

File: src/main/kotlin/extensions/mhw/Rollhunt.kt
package extensions.mhw

import arguments.MhwRollHunt
import com.kotlindiscord.kord.extensions.commands.application.slash.EphemeralSlashCommand
import com.kotlindiscord.kord.extensions.components.forms.ModalForm
import com.kotlindiscord.kord.extensions.types.respond
import dev.kord.common.Color
import dev.kord.rest.builder.message.create.embed
import kotlinx.datetime.Clock
import utils.CatBot
import java.util.logging.Logger


val MhwRollHuntCommand: suspend EphemeralSlashCommand<out MhwRollHunt, ModalForm>.() -> Unit = {
    name = "rollhunt"
    description = "Get a random roll of what monster you should hunt with which gear"

    action {
        val log = Logger.getLogger("Mhw")
        log.info("Received command: mhw %s (weapon=%s)".format(commandName, arguments.weaponType))

        val monster = ApiClient.MHW.monsters.random()
        val armor = ApiClient.MHW.armors.values.random()
        val weapon = ApiClient.MHW.weapons.values.filter {
            if (arguments.weaponType?.name == null) return@filter true
            it.type == arguments.weaponType?.readableName
        }.random()

        respond {
            embed {
                title = "Roll Hunt"
                color = Color.CatBot

                field {
                    name = "Monster"
                    value = monster.details.title
                }

                field {
                    name = "Armor"
                    value = armor.name
                }

                field {
                    name = "Weapon"
                    value = "${weapon.type}: ${weapon.name}"
                }

                timestamp = Clock.System.now()

                thumbnail {
                    url = monster.details.thumbnail
                }

                footer { text = "RollHunt" }
            }
        }
    }
}